test/qubit.jl

```diff
@@ -0,0 +1,270 @@
+using ITensors
+
+#
+# qubit
+#
+
+ITensors.space(::SiteType"qubit") = 2
```

Comment:
Should we consider moving the qubit site type into the library? Seems like it might get used a lot and we could publicize it as an out-of-the-box feature of ITensors. If we do, let's name it "Qubit" for better consistency with "Electron" and "Fermion".

Reply:
We can consider it, let's discuss it over Slack. Also agreed about changing it to "Qubit".

Reply:
Basically, at least at first I would lean towards providing it as an "example" rather than as an officially supported site type, since there are so many different gates that could be added which have different naming conventions.

Reply:
Changed to "Qubit", and I've moved `qubit.jl` to `examples/gate_evolution/` (and will hold off on making it an "official" site type for a little while once this gets tested a bit more).